### DISCUSSION TITLE
RavenDB with dynamic OID for databases

### DISCUSSION THREAD

**Discussion by mkocaks** [prometheus/snmp_exporter#884] [Q&A] [answered]
Hi,
After a few days of researching it seems it may not be possible for the following with snmp exporter for prometheus.
We are looking to monitor our RavenDB Cluster and all metrics via SNMP using SNMP Exporter with Prometheus, however the way RavenDB references the databases is problematic!
For each database the OID is different and we would rather not statically define in snmp.yaml each OID for each database (that can change in future if we remove and/or add databases to the raven cluster)
https://github.com/ravendb/ravendb/blob/v5.4/src/Raven.Server/Monitoring/Snmp/SnmpOids.cs

i.e,
[Description("Database name")]
            public const string Name = "5.2.{0}.1.1
            
 Is there a way we can dynamically walk through each unknown oid off the parent oid and get each database metrics with snmp exporter or is our only option to statically define these out of band and keep updating the snmp config each time we add or remove databases ?

**Comment by mkocaks** [comment]
This has been answered here:
https://github.com/ravendb/ravendb/discussions/16573

**Comment by SuperQ** [comment, marked as answer]
Yea, there's not much we can do here if the vendor doesn't follow SNMP spec/conventions. Looks like they implemented SNMP incorrectly.

**Reply by mkocaks** [reply]
My feeling is that the vendor implemented SNMP features focused around Zabbix which can handle this strange situation with custom scripts/macros on the Zabbix side... thank you for your reply and we have a better understanding of vendors implementation now which hard maps the dynamically generated OID's for each "database name" forever - which mitigates the risk of corrupting the TSDB in Prometheus (i.e. no risk of using the same OID ingested previously for a different database name)
So for now we are creating our own tooling to generate the snmp.yml from the vendors OID's (via our internal endpoint on the vendors side)

**Comment by SuperQ** [comment]
A quick search suggests that Telegraf -> Prometheus is a better option for this vendor anyway.

**Reply by mkocaks** [reply]
Yep but we need iop metrics which is missing from the plugin...

**Reply by mkocaks** [reply]
We have already looked at this, plus would be great not to have telegraf as man in the middle...
